warehouse/query-core/src/main/java/datawave/query/jexl/DatawavePartialInterpreter.java

```diff
@@ -0,0 +1,887 @@
+     *
+     * @param node
+     *            an ASTFunctionNode
+     * @param data
+     *            some data
+     * @return a result
+     */
+    @Override
+    public Object visit(ASTFunctionNode node, Object data) {
+        
+        // get the namespace
+        FunctionJexlNodeVisitor visitor = new FunctionJexlNodeVisitor();
+        visitor.visit(node, null);
+        
+        // content functions should always have the field as the first argument
+        if (visitor.namespace().equals(ContentFunctions.CONTENT_FUNCTION_NAMESPACE)) {
+            JexlNode first = visitor.args().get(0);
+            if (first instanceof ASTIdentifier) {
+                String field = JexlASTHelper.deconstructIdentifier((ASTIdentifier) first);
+                if (context.has(field) && incompleteFields.contains(field)) {
+                    return new State(true, true);
+                }
+            }
+        } else {
+            // fall back to descriptor for non-content functions
+            JexlArgumentDescriptor descriptor = JexlFunctionArgumentDescriptorFactory.F.getArgumentDescriptor(node);
+            Set<String> fields = descriptor.fields(null, null);
+            
+            for (String field : fields) {
+                if (context.has(field) && incompleteFields.contains(field)) {
+                    return new State(true, true);
+                }
+            }
+        }
+        
+        // if no determination could be made about an UNKNOWN status, fallback to a normal visit
+        Object result = super.visit(node, data);
+        if (result instanceof Integer || result instanceof Long) {
+            // filter functions like 'getMaxTime' or 'getMinTime'. No incomplete fields exist if we got to this point
+            return new State(true, false, new FunctionalSet(), result);
+        }
+        
+        if (result instanceof Collection) {
+            boolean isEmpty = ((Collection<?>) result).isEmpty();
+            if (hasSiblings(node)) {
+                if (isEmpty) {
+                    return new State(true, false);
+                } else {
+                    if (result instanceof FunctionalSet) {
+                        return new State(true, false, (FunctionalSet) result);
+                    } else {
+                        return new State(true, false, new FunctionalSet(), result);
+                    }
+                }
+            } else {
+                if (isEmpty) {
+                    return new State(true);
+                } else {
+                    return new State(false);
+                }
+            }
+        }
+        return getState(node, result);
+    }
+    
+    @Override
+    public Object visit(ASTEQNode node, Object data) {
+        String nodeString = JexlStringBuildingVisitor.buildQueryWithoutParse(node);
+        Object result = resultMap.get(nodeString);
+        if (null != result)
+            return new State(true); // TODO -- this might be from an unknown field
```

Comment:
Maybe check instanceof State and cast, or otherwise pass along all relevant State to the new object

Reply:
I don't think State objects belong in the resultMap, but I could be persuaded otherwise. 

The unspoken contract for the partial interpreter is to make a decision based on the field and the result of a normal interpreter visit.